crates/runtime/src/component/mod.rs

```diff
@@ -271,6 +365,70 @@ pub type InvocationStream = Pin<
     >,
 >;
 
+impl<H> Component<H>
+where
+    H: MinimalHandler,
+{
+    /// Compiles a WebAssembly component using [Runtime]. Modules are not supported with this function
+    ///
+    /// If you need to pass a custom context, use [CustomCtxComponent::new_with_linker_minimal].
+    ///
+    /// It is expected that the `linker_fn` is used to link any and all imports to the component. If
+    /// not, then the instantiation of the instance will fail.
+    #[instrument(level = "trace", skip_all)]
+    pub fn new_with_linker_minimal(
+        rt: &Runtime,
+        wasm: &[u8],
+        linker_fn: impl FnOnce(&mut Linker<Ctx<H>>) -> anyhow::Result<()>,
+    ) -> anyhow::Result<Self> {
+        if wasmparser::Parser::is_core_wasm(wasm) {
+            anyhow::bail!("core modules are not supported in the minimal linker");
+        }
+        let engine = rt.engine.clone();
+        let claims = None;
+        let component = wasmtime::component::Component::new(&engine, wasm)
+            .context("failed to compile component")?;
+
+        let mut linker = Linker::new(&engine);
+        linker_fn(&mut linker)?;
```

Comment:
Is there any reason why the linker function is called earlier here compared to the `CustomCtxComponent`? If there is a reason, we should probably have a comment here. If there isn't, maybe put it in the same place for consistency (consider this a nit)? This could be solved with the reusable bit

Reply:
no, but it definitely should be called later to ensure that our builtins go in first. Will refactor